lib/metrics.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+module Metrics
+  class TimeSeries
+    attr_accessor :labels, :query
+
+    def initialize(labels:, query:)
+      @labels = labels
+      @query = query
+    end
+  end
+
+  class MetricDefinition
+    attr_accessor :name, :description, :unit, :series
+
+    def initialize(name:, description:, unit:, series:)
+      @name = name
+      @description = description
+      @unit = unit
+      @series = series
+    end
+  end
+
+  POSTGRES_METRICS = [
+    MetricDefinition.new(
+      name: "CPU Usage",
```

Comment:
Should we change the metric name to something URL-safe, like `cpu_usage`? We can keep `CPU Usage` as the display name. Sending the `Cache hit ratio` string in the request doesn't feel very programmatic.

Reply:
I was thinking about this, along with the change to use the hash below. I will make this change.

Reply:
Done both